### PR TITLE
Remove compiler ras address masking

### PR DESCRIPTION
**Pull request by 0xdaryl**
"Address masking" was an attempt to make compiler log files easier to compare by preventing addresses (that are likely to change between compilations) from being printed raw in a compilation log.  It worked by intercepting the "printf" family of IO functions and checking whether any addresses were being printed with the `%p` format specifier, and instead of printing the address a "Masked Address" string was substituted instead.

This feature was introduced at a time when text file comparison tools were not as robust as they are today and many addresses are output to the log without using `%p` making this feature less effective.  Since it adds overhead and complexity to the logging path, has incomplete coverage, and is not widely used (if at all), remove it from the code base.

Removing address masking was discussed briefly at the Oct 24 OMR Architecture Meeting with agreement to remove it.

### CODE REVIEW

**Comment by 0xdaryl** [comment]
@JBKingdon @klangman : FYI for any opinions you might have

**Comment by 0xdaryl** [comment]
@dsouzai : would you mind reviewing please?

**Comment by JamesKingdon** [comment]
Sounds good to me.

File: compiler/compile/OMRCompilation.cpp [hunk 1]
```diff
@@ -2499,8 +2499,7 @@ void OMR::Compilation::diagnosticImplVA(const char *s, va_list ap)
       va_list copy;
       va_copy(copy, ap);
       char buffer[256];
-      TR::IO::vfprintf(self()->getOutFile(), self()->getDebug()->getDiagnosticFormat(s, buffer, sizeof(buffer)/sizeof(char)),
-                 copy);
+      TR::IO::vfprintf(self()->getOutFile(), s, copy);
```
Comment (dsouzai) [inline compiler/compile/OMRCompilation.cpp:2502]: Why did we ever need `copy`? Or this method for that matter; why couldn't any of this just be done in `OMR::Compilation::diagnosticImpl(const char *s, ...)`? Not sure if cleaning this up is part of a future PR in this series of PRs.

File: compiler/x/codegen/X86Debug.cpp [hunk 2]
```diff
@@ -620,12 +620,9 @@ TR_Debug::print(TR::FILE *pOutFile, TR::X86FenceInstruction  * instr)
          trfprintf(pOutFile, " External Absolute [");
       else
          trfprintf(pOutFile, " Relative [");
-
-      if (!_comp->getOption(TR_MaskAddresses))
-         {
          for (auto i = 0U; i < instr->getFenceNode()->getNumRelocations(); ++i)
             trfprintf(pOutFile," " POINTER_PRINTF_FORMAT, instr->getFenceNode()->getRelocationDestination(i));
```
Comment (dsouzai) [inline compiler/x/codegen/X86Debug.cpp:624]: nitpick: indentation; only mentioning it as this is not part of the `else` above.

**Review by dsouzai — COMMENTED** [review]
Overall LGTM aside from a couple of comments.

File: compiler/x/codegen/X86Debug.cpp (hunk 2 above)
Comment (0xdaryl) [reply compiler/x/codegen/X86Debug.cpp]: Fixed in forced push.

File: compiler/compile/OMRCompilation.cpp (hunk 1 above)
Comment (0xdaryl) [reply compiler/compile/OMRCompilation.cpp]: Fixed in forced push.

**Review by dsouzai — APPROVED** [review]

**Comment by dsouzai** [comment]
jenkins build all

**Comment by dsouzai** [comment]
win build failure:

```
[2024-11-19T14:41:35.197Z]          Debug.cpp
[2024-11-19T14:41:36.525Z]     32>c:\omr\workspace\pullrequest-win_x86-64\build\compiler\ras\debug.cpp(531): error C2220: warning treated as error - no 'object' file generated [C:\omr\workspace\PullRequest-win_x86-64\Build\build\jitbuilder\jitbuilder.vcxproj]
[2024-11-19T14:41:36.525Z]     32>c:\omr\workspace\pullrequest-win_x86-64\build\compiler\ras\debug.cpp(531): warning C4101: 'buffer': unreferenced local variable [C:\omr\workspace\PullRequest-win_x86-64\Build\build\jitbuilder\jitbuilder.vcxproj]
```

**Comment by 0xdaryl** [comment]
Jenkins build all

**Pull request merged**